plugins/utilities/natpmp_upnp.py

```diff
@@ -37,62 +41,162 @@ def wrapper(*args, **kwargs):
+    packages = {
+        "upnp-client": {
+            "url": "https://files.pythonhosted.org/packages/dd/69/4d38d9fa757c328df93e7037eb8c1da8ca48e62828c23ba3c421e9335e30/upnpclient-1.0.3.tar.gz",
+            "md5": "f936c8de89705555f6bd736a66d3af5d",
+            "folder": "upnpclient",
+        },
+        "python-dateutil": {
+            "url": "https://files.pythonhosted.org/packages/66/c0/0c8b6ad9f17a802ee498c46e004a0eb49bc148f2fd230864601a86dcf6db/python-dateutil-2.9.0.post0.tar.gz",
+            "md5": "81cb6aad924ef40ebfd3d62eaebe47c6",
+            "folder": "dateutil",
+        },
+        "six": {
+            "url": "https://files.pythonhosted.org/packages/71/39/171f1c67cd00715f190ba0b100d606d440a28c93c7714febeca8b79af85e/six-1.16.0.tar.gz",
+            "md5": "a7c927740e4964dd29b72cebfc1429bb",
+            "folder": "six.py",
+        },
+        "requests": {
+            "url": "https://files.pythonhosted.org/packages/9d/be/10918a2eac4ae9f02f6cfe6414b7a155ccd8f7f9d4380d62fd5b955065c3/requests-2.31.0.tar.gz",
+            "md5": "941e175c276cd7d39d098092c56679a4",
+            "folder": "requests",
+        },
+        "charset_normalizer": {
+            "url": "https://files.pythonhosted.org/packages/63/09/c1bc53dab74b1816a00d8d030de5bf98f724c52c1635e07681d312f20be8/charset-normalizer-3.3.2.tar.gz",
+            "md5": "0a4019908d9e50ff13138e8a794d9e2b",
+            "folder": "charset_normalizer",
+        },
+        "idna": {
+            "url": "https://files.pythonhosted.org/packages/bf/3f/ea4b9117521a1e9c50344b909be7886dd00a519552724809bb1f486986c2/idna-3.6.tar.gz",
+            "md5": "70f4beef4feb196ac64b75a93271f53c",
+            "folder": "idna",
+        },
+        "urllib3": {
+            "url": "https://files.pythonhosted.org/packages/7a/50/7fd50a27caa0652cd4caf224aa87741ea41d3265ad13f010886167cfcc79/urllib3-2.2.1.tar.gz",
+            "md5": "872f7f43af1b48e7c116c7542ab39fab",
+            "folder": "urllib3",
+        },
+        "ifaddr": {
+            "url": "https://files.pythonhosted.org/packages/e8/ac/fb4c578f4a3256561548cd825646680edcadb9440f3f68add95ade1eb791/ifaddr-0.2.0.tar.gz",
+            "md5": "b1cac02b5dc354d68dd6d853bc9565a7",
+            "folder": "ifaddr",
+        },
+        "NAT-PMP": {
+            "url": "https://files.pythonhosted.org/packages/dc/0c/28263fb4a623e6718a179bca1f360a6ae38f0f716a6cacdf47e15a5fa23e/NAT-PMP-1.3.2.tar.gz",
+            "md5": "7e5faa22acb0935f75664e9c4941fda4",
+            "folder": "natpmp",
+        },
+    }
+
+    system = platform.platform()
+    if "Windows" in system:
+        packages["lxml"] = {
+            "url": "https://files.pythonhosted.org/packages/fe/71/700f9baa762fee6dd63db8b16d78aeb4fc27aa2866c7b1c69c93c1e715e9/lxml-5.1.0-cp311-cp311-win32.whl",
+            "md5": "a9486faba8dd3c4caa8c20a0f699958b",
+            "folder": "lxml",
+        }
+    elif "Darwin" and "arm64" in system:

```

Comment:
should be `elif "Darwin" in system and "arm64" in system:`
needs similar changes other places too